async-nats/src/jetstream/stream.rs

```diff
@@ -331,6 +350,36 @@ pub struct StreamState {
     pub consumer_count: usize,
 }
 
+/// A raw stream message in the representation it is stored.
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct Message {
+    /// Subject of the message.
+    #[serde(rename = "subject")]
+    pub subject: String,
+
+    /// Sequence of the message.
+    #[serde(rename = "seq")]
+    pub sequence: u64,
+
+    /// Data of the mssage.
+    #[serde(default, rename = "data")]
+    pub data: String,
+
+    /// Raw header string, if any.
+    #[serde(default, rename = "hdrs")]
+    pub headers: Option<String>,
+
+    /// The time the message was published.
+    #[serde(rename = "time", with = "rfc3339")]
+    pub time: time::OffsetDateTime,
+}
+
+/// Helper type to deserialize and flatten message requests
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct MessageStatus {
```

Comment:
Ideally, probably flatten this

```suggestion
pub struct MessageStatus {
```